core/chains/evm/gas/chainoracles/l1_gas_price_oracle.go

```diff
@@ -0,0 +1,168 @@
+package chainoracles
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+//go:generate mockery --quiet --name ethClient --output ./mocks/ --case=underscore --structname ETHClient
+type ethClient interface {
+	CallContract(ctx context.Context, msg ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+}
+
+// Reads L2-specific precompiles and caches the l1GasPrice set by the L2.
+type l1GasPriceOracle struct {
+	client     ethClient
+	pollPeriod time.Duration
+	logger     logger.Logger
+	address    string
+	selector   string
+
+	l1GasPriceMu sync.RWMutex
+	l1GasPrice   *assets.Wei
+
+	chInitialised chan struct{}
+	chStop        utils.StopChan
+	chDone        chan struct{}
+	utils.StartStopOnce
+}
+
+const (
+	// ArbGasInfoAddress is the address of the "Precompiled contract that exists in every Arbitrum chain."
+	// https://github.com/OffchainLabs/nitro/blob/f7645453cfc77bf3e3644ea1ac031eff629df325/contracts/src/precompiles/ArbGasInfo.sol
+	ArbGasInfoAddress = "0x000000000000000000000000000000000000006C"
+	// ArbGasInfo_getL1BaseFeeEstimate is the a hex encoded call to:
+	// `function getL1BaseFeeEstimate() external view returns (uint256);`
+	ArbGasInfo_getL1BaseFeeEstimate = "f5d6ded7"
+
+	// GasOracleAddress is the address of the precompiled contract that exists on OP stack chain.
+	// This is the case for Optimism and Base.
+	OPGasOracleAddress = "0x420000000000000000000000000000000000000F"
+	// GasOracle_l1BaseFee is the a hex encoded call to:
+	// `function l1BaseFee() external view returns (uint256);`
+	OPGasOracle_l1BaseFee = "519b4bd3"
+
+	// Interval at which to poll for L1BaseFee. A good starting point is the L1 block time.
+	PollPeriod = 12 * time.Second
+)
+
+func NewL1GasPriceOracle(lggr logger.Logger, ethClient ethClient, chainType config.ChainType) L1Oracle {
+	var address, selector string
+	switch chainType {
+	case config.ChainArbitrum:
+		address = ArbGasInfoAddress
+		selector = ArbGasInfo_getL1BaseFeeEstimate
+	case config.ChainOptimismBedrock:
+		address = OPGasOracleAddress
+		selector = OPGasOracle_l1BaseFee
+	default:
+		return nil
+	}
+
+	return &l1GasPriceOracle{
+		client:        ethClient,
+		pollPeriod:    PollPeriod,
+		logger:        lggr.Named(fmt.Sprintf("L1GasPriceOracle(%s)", chainType)),
+		address:       address,
+		selector:      selector,
```

Comment:
How about `call` instead, since this is a hex-encoded call? Selector doesn't provide much context.